presidio-analyzer/analyzer/analyzer_engine.py

```diff
@@ -108,6 +118,10 @@ def analyze(self, text, entities, language, all_fields):
         # run the nlp pipeline over the given text, store the results in
         # a NlpArtifacts instance
         nlp_artifacts = self.nlp_engine.process_text(text, language)
+
+        self.ml_tracer.trace(request_id,
+                             "nlp artifacts:" + repr(nlp_artifacts))
```

Comment:
Better to use the \_\_repr\_\_ function inside nlp_artifacts. This is prone to errors.